sound/soc/intel/boards/sof_sdw.c

```diff
@@ -899,8 +899,15 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 				.widgets = generic_spk_widgets,
 				.num_widgets = ARRAY_SIZE(generic_spk_widgets),
 			},
+			{
+				.direction = {false, true},
+				.dai_name = "rt712-sdca-aif3",
+				.dai_type = SOF_SDW_DAI_TYPE_MIC,
+				.dailink = {SDW_UNUSED_DAI_ID, SDW_DMIC_DAI_ID},
+				.rtd_init = rt_dmic_rtd_init,
+			},
 		},
-		.dai_num = 2,
+		.dai_num = 3,
```

Comment:
rt713 has -vb version, too.
```
diff --git a/sound/soc/intel/boards/sof_sdw.c b/sound/soc/intel/boards/sof_sdw.c
index 87e90e9a90a8..8fd02af61dd4 100644
--- a/sound/soc/intel/boards/sof_sdw.c
+++ b/sound/soc/intel/boards/sof_sdw.c
@@ -951,8 +951,15 @@ static struct sof_sdw_codec_info codec_info_list[] = {
                                .widgets = generic_jack_widgets,
                                .num_widgets = ARRAY_SIZE(generic_jack_widgets),
                        },
+                       {
+                               .direction = {false, true},
+                               .dai_name = "rt712-sdca-aif3",
+                               .dai_type = SOF_SDW_DAI_TYPE_MIC,
+                               .dailink = {SDW_UNUSED_DAI_ID, SDW_DMIC_DAI_ID},
+                               .rtd_init = rt_dmic_rtd_init,
+                       },
                },
-               .dai_num = 1,
+               .dai_num = 2,
        },
        {
                .part_id = 0x1713,
```

Reply:
that's what I asked earlier, I don't understand which part does what :-)

Reply:
> that's what I asked earlier, I don't understand which part does what :-)

The -va version uses a separated sdw interface (rt1712/rt1713) for DMIC, and -vb uses the same sdw interface. -vb = -va + dmic, rt1713-va/rt1712-va = dmic.
And, rt713 = rt712 - amp,  
So, rt712-va = jack + amp, rt713-va = jack, rt712-vb = jack + amp + dmic, rt713-vb = jack + dmic, rt1713-va/rt1712-va = dmic